web/TrackPlayer/Player.ts

```diff
@@ -1,6 +1,8 @@
 import { State } from '../../src/constants/State';
 import type { Track, Progress, PlaybackState } from '../../src/interfaces';
 import { SetupNotCalledError } from './SetupNotCalledError';
+// @ts-ignore
+import shaka from 'shaka-player/dist/shaka-player.ui';
```

Comment:
@Bram-dc can you please explain how this is intended to work? Specifically, how does it work in SSR (e.g. Next.js). I'd also like to understand the motivation behind the change.

Thanks!

Reply:
I have not used Next.js with react native. Is that possible? React-native-track-player has to be on the client right? What has SSR to do with it?

Reply:
@Bram-dc yes, SSR for web is definitely possible, but in SSR contexts you can't import or execute JS packages that rely on the dom or other things dependent upon the browser. In order to facilitate this, you dynamically import the module so that it's only imported on the clientside and not on the server.

I'm open to a better method of doing this, but I'm not open to removing SSR support. If there's a way we can achieve both (which I'm not aware of) I'd love to learn what it is.

Reply:
The example app is working after this change.

Reply:
I also want to be clear, that the `example` web project is just a basic SPA react app from create-react-app. Meaning, it doesn't do SSR or contain the SSR functionality, so if you're trying to see if your change works on SSR using the example isn't going to do it for you.

Reply:
What part of this project would be running on the server, before serving to the client? Would the RNTP package not be imported as a client package entirely when using it in a SSR project?

Reply:
@Bram-dc typically, there are a lot of UI elements that are pretty deeply tied to RNTP functionality (event hooks, helpers, etc). In order for people to adopt web into their projects, they'd likely need to invest a lot of effort in refactoring their UI so that it's based on a dynamically imported version of RNTP. Additionally, that makes things bit tricky, in that, they're also writing code that supports iOS and Android and that same code is deployed across all platforms. I'm open to being convinced otherwise, but asking users to dynamically import RNTP only in web so they can use SSR seems like a challenging proposition. It might be a different story for people who are just getting started with their UI. I'm also not clear on the problems the dynamic import is causing for you, so it's hard to see the advantages or the need.

Again, this is an early feature set so I'm sure there's a lot I haven't considered in it and I'm open to changing it for the right reasons. I just need some more help understanding the challenges you're having and balancing them with the tradeoffs.

Reply:
@Bram-dc , thoughts here?